instrumentation-genai/opentelemetry-instrumentation-vertexai/tests/test_function_calling.py

```diff
@@ -0,0 +1,406 @@
+import pytest
+from vertexai.generative_models import (
+    Content,
+    FunctionDeclaration,
+    GenerativeModel,
+    Part,
+    Tool,
+)
+
+from opentelemetry.instrumentation.vertexai import VertexAIInstrumentor
+from opentelemetry.sdk._logs._internal.export.in_memory_log_exporter import (
+    InMemoryLogExporter,
+)
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import (
+    InMemorySpanExporter,
+)
+
+
+@pytest.mark.vcr
+def test_function_call_choice(
+    span_exporter: InMemorySpanExporter,
+    log_exporter: InMemoryLogExporter,
+    instrument_with_content: VertexAIInstrumentor,
+):
+    ask_about_weather()
+
+    # Emits span
+    spans = span_exporter.get_finished_spans()
+    assert len(spans) == 1
+    assert spans[0].name == "chat gemini-1.5-flash-002"
+    assert dict(spans[0].attributes) == {
+        "gen_ai.operation.name": "chat",
+        "gen_ai.request.model": "gemini-1.5-flash-002",
+        "gen_ai.response.finish_reasons": ("stop",),
+        "gen_ai.response.model": "gemini-1.5-flash-002",
+        "gen_ai.system": "vertex_ai",
+        "gen_ai.usage.input_tokens": 72,
+        "gen_ai.usage.output_tokens": 16,
+        "server.address": "us-central1-aiplatform.googleapis.com",
+        "server.port": 443,
+    }
+
+    # Emits user and choice events
+    logs = log_exporter.get_finished_logs()
+    assert len(logs) == 2
+    user_log, choice_log = [log_data.log_record for log_data in logs]
+    assert user_log.attributes == {
+        "gen_ai.system": "vertex_ai",
+        "event.name": "gen_ai.user.message",
+    }
+    assert user_log.body == {
+        "content": [
+            {"text": "Get weather details in New Delhi and San Francisco?"}
+        ],
+        "role": "user",
+    }
+
+    assert choice_log.attributes == {
+        "gen_ai.system": "vertex_ai",
+        "event.name": "gen_ai.choice",
+    }
+    assert choice_log.body == {
+        "finish_reason": "stop",
+        "index": 0,
+        "message": {
+            "content": [
+                {
```

Comment:
Shouldn't these be inside the `tool_calls` key in `message`?

Reply:
Yes, it's an issue with the semantic conventions. I opened https://github.com/open-telemetry/semantic-conventions/issues/1883.

I also left some inline comments in `utils.py` explaining why I chose this for now, lmk if you have any other idea.

Reply:
Created a contrib issue to track this as well https://github.com/open-telemetry/opentelemetry-python-contrib/issues/3280